### DISCUSSION TITLE
How to show info regarding in-chunk slow processes

### DISCUSSION THREAD

**Discussion by bakaburg1** [quarto-dev/quarto-cli#9926] [Q&A] [answered]
### Description

Hello,

Some chunks have long-running operations where the quarto progress bar for the overall document just stops until the chunk code is fully executed.
Is there a way to show some info produced by the chunk code so that one can understand if it's progressing? 
For example some logging messages or even better a progress bar.

My language of reference is R.

**Comment by mcanouil** [comment]
R code is ran via `knitr`. That's the engine and it's it that displays the progression.

See <https://yihui.org/en/2023/01/knitr-progress-bar/> and the `knitr.progress.fun` option (<https://yihui.org/knitr/options/#global-r-options>.

**Reply by bakaburg1** [reply]
uhm, I'm not sure I follow. That just edit how the progress bar for the whole document is displayed isn't it? it doesn't allow to get additional logging messages or progress bars for the code inside one chunck.

**Reply by mcanouil** [reply]
Quarto does not control that.
This is up to you to add messages etc. in the code cells and/or customise the progress function to display more information.

Quarto or knitr do not mangle with your code.

**Reply by bakaburg1** [reply]
yeah, but messages and other logging outputs are not displayed when I process the document!

**Reply by mcanouil** [reply]
Do you have a concrete example?

**Reply by mcanouil** [reply, marked as answer]
Still, that's a `knitr` option to set.
If you don't want the messages to be printed in the rendered document, but in the console, the `knitr` documentation says:

```
knitr::opts_chunk$set(message = NA)
```

The same goes for warnings.

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/aa358c01-3b77-4b0c-9ca1-a1b5c3f40dc8">

(Note that I am using a custom progress bar.)

**Reply by bakaburg1** [reply]
that's it!! I was missing the `message = NA` option!
